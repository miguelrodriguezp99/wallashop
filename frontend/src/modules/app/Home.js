import React, { useContext, useEffect } from "react";
import { config } from "../../config/constants.js";
import { LogginContext } from "./context/loginContext.js";
import "./styles/home.css";
import { Filters } from "./components/Filters.js";
import Posts from "./components/Posts.js";
import CreatePostModal from "./components/modals/CreatePostModal.js";
import useStomp from "./hooks/useStomp.js";
import NotificationsReload from "./components/NotificationsReload.js";

export default function Home() {
	let { token, setToken, setUser } = useContext(LogginContext);

	useEffect(() => {
		const bearer = localStorage.getItem(config.SERVICE_TOKEN_NAME);
		const user = localStorage.getItem("user");
		if (bearer !== null) {
			setToken(bearer);
			setUser(JSON.parse(user));
		}
	}, [setToken, setUser]);

	const { isNotified } = useStomp(token);

	return (
		<>
			{ 
			isNotified === true 
			? 
				<NotificationsReload />
			: 
				null
				}
			<div className="home" >
				<div className="content">
					<h1 className="text-7xl font-bold bg-gradient-to-r from-blue-500 to-blue-700 text-transparent bg-clip-text">
						Our Articles
					</h1>
					<Filters />
					<div className="posts-container">
						<Posts />
						{token != null ? <CreatePostModal /> : null}
					</div>
				</div>
			</div>
		</>
	);
}
