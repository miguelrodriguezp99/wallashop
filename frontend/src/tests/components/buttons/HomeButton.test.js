import React from 'react';
import Home from '../../../modules/app/components/buttons/HomeButtons';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';
import { Route, Routes } from 'react-router-dom';

describe('Home component', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(
        <MemoryRouter>
          <Home text="Home" route="/home" />
        </MemoryRouter>
      );
  
      expect(asFragment()).toMatchSnapshot();
    });
      
    it('navigates to the correct route', async () => {
        const { getByText } = render(
          <MemoryRouter initialEntries={['/']}>
            <Routes>
              <Route path="/" element={<Home text="Home" route="/home" />} />
              <Route path="/home" element={<div>text</div>} />
            </Routes>
          </MemoryRouter>
        );
      
        fireEvent.click(getByText('Home'));
        await waitFor(() => expect(screen.getByText('text')).toBeInTheDocument());
      });
  });