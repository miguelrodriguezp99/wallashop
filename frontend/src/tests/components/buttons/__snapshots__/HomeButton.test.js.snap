// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="lgn-btn"
    type="submit"
  >
    Home
  </button>
</DocumentFragment>
`;
